source/Coop/NetImpl/LiteNet/DiscoveryThread.cs

```diff
@@ -0,0 +1,133 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Threading;
+using Common;
+using JetBrains.Annotations;
+using LiteNetLib;
+using Network;
+using Network.Infrastructure;
+using Network.Protocol;
+using NLog;
+
+namespace Coop.NetImpl.LiteNet
+{
+    /// <summary>
+    ///     Thread to discovery coop servers in LAN.
+    /// </summary>
+    public class DiscoveryThread
+    {
+        private static readonly Logger Logger = LogManager.GetCurrentClassLogger();
+        private readonly object m_AvailableServerLock = new object();
+
+        private readonly HashSet<IPEndPoint> m_AvailableServers = new HashSet<IPEndPoint>();
+
+        private readonly NetworkConfiguration m_Config;
+
+        private readonly NetManager m_NetManager;
+        private readonly object m_StopRequestLock = new object();
+        [CanBeNull] private FrameLimiter m_FrameLimiter;
+        private bool m_IsStopRequest;
+        private Thread m_Thread;
+
+        /// <summary>
+        ///     Will immediately spawn a new thread and start listing for discovery packets from
+        ///     bannerlord coop servers.
+        /// </summary>
+        /// <param name="config"></param>
+        public DiscoveryThread([NotNull] NetworkConfiguration config)
```

Comment:
A Discovery Task would be more compute efficiant. This works for now but I will accept this and create an issue (#106)